web/src/lib/components/asset-viewer/editor/crop-canvas.svelte

```diff
@@ -0,0 +1,740 @@
+<script lang="ts">
+  import {
+    cropAspectRatio,
+    cropImageScale,
+    cropImageSize,
+    cropSettings,
+    type CropSettings,
+    type CropAspectRatio,
+    cropSettingsChanged,
+  } from '$lib/stores/asset-editor.store';
+  import { onMount, afterUpdate } from 'svelte';
+  import { get } from 'svelte/store';
+
+  export let crop = get(cropSettings);
+  export let asset;
+  export let aspectRatio = get(cropAspectRatio);
+  cropSettings.subscribe((value) => {
+    crop = value;
+  });
+
+  let canvas: HTMLCanvasElement;
+  let ctx: CanvasRenderingContext2D;
+  let img: HTMLImageElement;
+
+  let darkenLevel = 0.65; // Initial darkening level
+  let animationFrame: ReturnType<typeof requestAnimationFrame>;
+  let canvasCursor: string;
+  const getAssetUrl = (id: string, checksum: string) => {
+    return `http://localhost:2283/api/assets/${id}/original?c=${checksum}`;
```

Comment:
What if the original asset can't be displayed by the browser? Also this URL should be changed

Reply:
This is only partially resolved. There is still the possibility that a browser can't display the original asset. This will also be an issue for videos, live photos and panoramas.

Reply:
Should we even allow the editor to be opened for live photos, panoramas, and gifs?

Reply:
Right now it makes sense to disable the editor for these types of assets. If there are tools added in the future that can be used for more asset types (like rotate), we can always reconsider this

Reply:
@michelheusschen 
>  There is still the possibility that a browser can't display the original asset

And then what to show? now there is an error message when opening the editor and an error loading the image. An error message, a black editor screen, and a toolbar on the right. 
Should I close the editor automatically in this case?

Reply:
> Right now it makes sense to disable the editor for these types of assets

done